lib/active_model/serializer/adapter/json.rb

```diff
@@ -47,6 +31,32 @@ def serializable_hash(options = {})
       def fragment_cache(cached_hash, non_cached_hash)
         Json::FragmentCache.new().fragment_cache(cached_hash, non_cached_hash)
       end
+
+      private
+
+      def populate_hash_for_array_serializer(name, association, opts)
+        return unless association.respond_to?(:each)
```

Comment:
The reason I moved it to a private method was to eliminate the nested conditionals in the iterator. I think the code is easier to read this way, and moving the guard condition back into the iterator defeats the purpose of this PR.